src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,165 @@
+                //   [assembly: System.Reflection.AssemblyCompany("Test")]
+                while (syntaxFactsService.IsElementAccessExpression(node.Parent))
+                {
+                    node = node.Parent;
+                }
+
+                // node -> AttributeArgumentSyntax -> AttributeArgumentListSyntax -> AttributeSyntax
+                var attributeSyntaxNodeCandidate = node.Parent?.Parent?.Parent;
+                if (syntaxFactsService.IsAttribute(attributeSyntaxNodeCandidate))
+                {
+                    return attributeSyntaxNodeCandidate;
+                }
+            }
+
+            return null;
+        }
+
+        private static async Task<bool> CheckTypeInfoOfAttributeAsync(CompletionContext context, SyntaxNode attributeNode)
+        {
+            var semanticModel = await context.Document.GetSemanticModelForNodeAsync(attributeNode, context.CancellationToken).ConfigureAwait(false);
+            var typeInfo = semanticModel.GetTypeInfo(attributeNode);
+            var type = typeInfo.Type;
+            if (type == null)
+            {
+                return false;
+            }
+
+            var internalsVisibleToAttributeSymbol = semanticModel.Compilation.GetTypeByMetadataName(typeof(InternalsVisibleToAttribute).FullName);
+            return type.Equals(internalsVisibleToAttributeSymbol);
+        }
+
+        private static async Task AddAssemblyCompletionItemsAsync(CompletionContext context, CancellationToken cancellationToken)
+        {
+            var currentProject = context.Document.Project;
+            var currentCompilation = await currentProject.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var currentAssemblyInfo = currentCompilation.Assembly;
+            foreach (var project in context.Document.Project.Solution.Projects)
+            {
+                if (project == currentProject)
+                {
+                    continue;
+                }
+
+                if (await HasProjectAlreadyAccessToInternals(currentAssemblyInfo, project, cancellationToken).ConfigureAwait(false))
+                {
+                    continue;
+                }
+
+                var projectGuid = project.Id.Id.ToString();
+                var completionItem = CommonCompletionItem.Create(
+                    displayText: project.AssemblyName,
+                    rules: CompletionItemRules.Default,
+                    glyph: project.GetGlyph(),
+                    properties: ImmutableDictionary.Create<string, string>().Add(ProjectGuidKey, projectGuid));
+                context.AddItem(completionItem);
+            }
+        }
+
+        private static async Task<bool> HasProjectAlreadyAccessToInternals(IAssemblySymbol sourceAssembly, Project project, CancellationToken cancellationToken)
+        {
+            var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Yeah, don't do this :). We added GetSemanticModelForNodeAsync because that made a difference compared to rebuilding semantics for the entire Document. I shudder to think what compiling every project in the sln will do. Imagine a solution like Roslyn with 200 projects...

Can we try your alternative solution?